### DISCUSSION TITLE
Do I actually need an ignition-controlled power-off/on trigger?

### DISCUSSION THREAD

**Discussion by Compizfox** [opencardev/crankshaft#462] [Q&A] [answered]
Hi all,

On the (old) wiki, [it is described](https://github.com/opencardev/crankshaft/wiki/Boot,-reboot-and-shutdown-the-Pi-with-ignition-key) how to build a circuit that powers off the RPi gracefully when the ignition is turned off using a relay*.

Unfortunately, this becomes problematic when the I2C bus is in use because GPIO3, which wakes the RPi from halt state, is also shared with the I2C SCL pin.

I suppose it is possible to circumvent this problem by only pulling GPIO3 low momentarily, but I was wondering, is all this even necessary? I was thinking it would be much easier to just connect the RPi's power supply to the ignition. Crankshaft mounts the filesystems RO, so file corruption from sudden power loss shouldn't be a concern, right? Am I missing something?

Of course, the drawback of this is that there is no shutdown time-out: the RPi immediately powers off when you remove the key from the ignition. But aside from this, I don't see any problem, assuming file corruption indeed isn't a concern. 

\* By the way, I think a relay is completely overkill since we're not switching large currents here. We just need to pull a GPIO pin low, so a simple BJT suffices. Even if galvanic isolation is a requirement (which I don't believe it is, since the RPi and the car share a common ground), an optocoupler would suffice.

**Comment by Compizfox** [comment, marked as answer]
OK, so I've been using it this way for a while now and it appears to be totally fine.

The only downside is that there is no time-out, so when you turn the ignition from ACC to off, or start the engine from ACC, the RPi will immediately reset. This is annoying if you want to e.g. setup your navigation before starting the engine.

**Comment by egisz** [comment]
In my case BMW F10, I hooked to radio 12v, which  shuts down after 30 min.
car is locked. Also had 0 issues.

**Reply by Compizfox** [reply]
That's an interesting idea! I don't know if my car has a 12V line like that (with delayed power off), but if it does, that's an attractive solution.
